data_engineering/pipeline_runner.py

```diff
@@ -1,71 +1,60 @@
 import pandas as pd
 
-from utils import feature_imputation, pipeline_io
-
-from utils.other_derivation import (
-    derive_total_assets_from_log_total_assets,
-    derive_short_term_securities_from_cash,
-    rename_column,
+from utils import pipeline_io
+from utils.feature_engineering import (
+    remove_companies_with_many_null_values,
+    remove_duplicates,
+    fill_null_with_zeros,
+    get_raw_values,
 )
-from utils.fourth_layer_derivation import derive_fourth_layer
-from utils.second_layer_derivation import derive_second_layer
-from utils.third_layer_derivation import derive_third_layer
 
+TRAIN = "train"
+TEST = "test"
 COLS_IMPUTE_ZEROS = ["gross profit (in 3 years) / total assets"]
 
 
-def get_raw_values(ratio_df):
-    """
-    - First and fifth layers are only deriving one column
-    - Exclude rotation receivables + inventory turnover in days since they are derived from receivable days and
-    inventory days which we can calculate from raw values
-
-    Args:
-        ratio_df:
-
-    Returns: Dataframe with 32 columns of raw values
-
-    """
-    raw_values_df = (
-        ratio_df.pipe(derive_total_assets_from_log_total_assets)
-        .pipe(derive_second_layer)
-        .pipe(derive_third_layer)
-        .pipe(derive_fourth_layer)
-        .pipe(derive_short_term_securities_from_cash)
-        .pipe(rename_column)
+def transform_ratio(df: pd.DataFrame) -> pd.DataFrame:
+    # todo: to be expanded
+    return (
+        df.pipe(fill_null_with_zeros, columns=COLS_IMPUTE_ZEROS)
+        .pipe(remove_duplicates)
+        .pipe(remove_companies_with_many_null_values, thres=55)
     )
 
-    return raw_values_df.drop(ratio_df.columns, axis=1)
 
-
-def transform(df: pd.DataFrame) -> pd.DataFrame:
-    df = feature_imputation.fill_null_with_zeros(df, COLS_IMPUTE_ZEROS)
-    return df
+def transform_raw(df: pd.DataFrame) -> pd.DataFrame:
+    # todo: to be expanded
+    return (
+        df.pipe(get_raw_values)
+        .pipe(remove_duplicates)
+        .pipe(remove_companies_with_many_null_values, thres=25)
+    )
 
 
-def run(clean_ratio=True, get_raw=True):
+def run():
     """
     Run two different jobs
     1. Clean the original ratio df
     2. Get the raw values from original ratio df
     """
-    pipeline_io.create_output_dir()
-    train_df = pipeline_io.read_train_file()
-    print(train_df.head())
+    output_files = {
+        "ratio_train": "ratio_train",
+        "ratio_test": "ratio_test",
+        "raw_train": "raw_train",
+        "raw_test": "raw_test",
+    }
 
-    if clean_ratio:
-        transformed_df = transform(train_df)
-
-        pipeline_io.save_clean_train_file(transformed_df)
-
-    if get_raw:
-        raw_values_df = get_raw_values(train_df)
+    pipeline_io.create_output_dir()
 
-        # the raw values df should have 32 features columns and 1 target column
-        assert raw_values_df.shape[1] == 32 + 1
+    df = pipeline_io.read_file(TRAIN)
```

Comment:
would it be better to pass TRAIN or TEST as a parameter to run() so we don't have to manually replace TRAIN here and below (when getting the file name from output_files dictionary)?

Reply:
agree

Reply:
my initial idea was to run both TRAIN and TEST in `run` in a for loop format, looping and processing each file, but some engineering steps couldn't be applied to TEST yet so i left that out. we can also do passing parameters if that's better